modules/loadbalancer/outputs.tf

```
@@ -4,6 +4,31 @@ output "backend_pool_id" {
 }
 
 output "frontend_ip_configs" {
-  value       = { for k, v in azurerm_lb.lb.frontend_ip_configuration : v.name => coalesce(try(data.azurerm_public_ip.exists[v.name].ip_address, ""), try(azurerm_public_ip.this[v.name].ip_address, ""), v.private_ip_address) }
-  description = "The Frontend configs of the loadbalancer."
+  value       = local.output_ips
+  description = "Map of IP addresses, one per each entry of `frontend_ips` input. Contains public IP address for the frontends that have it, private IP address otherwise."
+}
+
+output "frontend_combined_rules" {
+  value       = local.output_rules
+  description = <<-EOF
+  Map of all rules of all load balancer's frontends combined.
+  The map entries are intended to be easily convertable into an NSG rule, hence each entry
+  contains `port`, `protocol`,  `frontend_ip`, and numerical sequential `index`.
+  The `frontend_ip` is the same as as returned by output `frontend_ip_configs`.
+
+  Full example:
+
+  ```hcl
+  {
+    "frontend01-balancessh" = {
+      "fipkey" = "frontend01"
+      "frontend_ip" = "34.34.34.34"
+      "index" = 0
+      "port" = 22
+      "protocol" = "tcp"
+      "rulekey" = "balancessh"
+    }
+  }
```

```suggestion
{
  "frontend01-balancessh" = {
    "fipkey"      = "frontend01"
    "frontend_ip" = "34.34.34.34"
    "index"       = 0
    "port"        = 22
    "protocol"    = "tcp"
    "rulekey"     = "balancessh"
  }
}
```

---

Yikes, leftmost columns 1 and 2 break the `<<-EOF` convention, but I see what you mean:

```suggestion
  {
    "frontend01-balancessh" = {
      "fipkey"      = "frontend01"
      "frontend_ip" = "34.34.34.34"
      "index"       = 0
      "port"        = 22
      "protocol"    = "tcp"
      "rulekey"     = "balancessh"
    }
  }
```